glossary.adoc

```diff
@@ -0,0 +1,17 @@
+= Glossary of common terms within MicroProfile
+
+[options="header"]
+|===
+| Term | Description
+| Evolution process | A formal process to evolve what is included the MicroProfile
+| Proposal | A proposal for a new addition to the MicroProfile (a formal document describing what should be added and why)
+| Candidate | The work in process after a new proposal within the evolution process was accepted, and before the proposal comes to a final version so that it can become part of MicroProfile. May also refer to the repository related to an accepted proposal. Should include the definition of the proposed API or behavior. May also include a reference implementation and a compatibility kit
```

Comment:
@kenfinnigan proposed in the [mailing list](https://groups.google.com/d/msg/microprofile/j9urpwSd6es/YdSLbZDbEAAJ):

> I would lean towards "concept", as "candidate" can often have meanings associated with a near release. ie. release candidate.